### ISSUE TITLE
[Bug] MarkerProps "onClick" property should be optional

### ISSUE THREAD

**Issue by slaymance** [visgl/react-map-gl#1788]
### Description

The `MarkerProps` type definition for the [onClick()](https://github.com/visgl/react-map-gl/blob/master/src/components/marker.ts#L70) property is currently defined as required while documentation implies this property is optional.

### Expected Behavior

`onClick()` should be defined as optional in the `MarkerProps` type definition.

### Steps to Reproduce

Using `MarkProps` to type a wrapper for `Marker` will cause a compile error without an `onClick()` property.

```tsx
import { Marker } from 'react-map-gl';
import type { MarkerProps } from 'react-map-gl';

const MarkerWrapper = ({ latitude, longitude }: MarkerProps) => (
  <Marker latitude={latitude} longitude={longitude}>
    {props.children}
  </Marker>
);

/**
 * Result: Failed to compile.
 * Type error: Property 'onClick' is missing in type '{ longitude: number; latitude: number; }' but required in type 'MarkerProps'.
 */
```

### Environment

- Framework version: react-map-gl@7.0.8
- Map library: N/A
- Browser: N/A
- OS: N/A

### Logs

N/A

**Comment by slaymance** [comment]
Fix: https://github.com/visgl/react-map-gl/pull/1789

**Comment by Pessimistress** [comment]
Published 7.0.9
